server/blob/blob-objectstorage/src/test/java/org/apache/james/blob/objectstorage/BlobExistenceTesterIntegrationContract.java

```diff
@@ -0,0 +1,235 @@
+
+        assertThat(blobExistenceTester().exists(ObjectStorageBucketName.of(defaultBucketName), blobIdCustom).block())
+            .isTrue();
+    }
+
+    @Test
+    default void deleteShouldNotDeleteOtherBlobIdsFromOtherBucketsInBlobExistenceTester() {
+        BucketName defaultBucketName = testee().getDefaultBucketName();
+
+        BlobId blobIdDefault = testee().save(defaultBucketName, SHORT_BYTEARRAY).block();
+        BlobId blobIdCustom = testee().save(CUSTOM, OTHER_SHORT_BYTEARRAY).block();
+
+        testee().delete(defaultBucketName, blobIdDefault).block();
+
+        assertThat(blobExistenceTester().exists(ObjectStorageBucketName.of(CUSTOM), blobIdCustom).block())
+            .isTrue();
+    }
+
+    @Test
+    default void deleteShouldNotDeleteSameBlobIdsFromOtherBucketsInBlobExistenceTester() {
+        BucketName defaultBucketName = testee().getDefaultBucketName();
+
+        BlobId blobIdDefault = testee().save(defaultBucketName, SHORT_BYTEARRAY).block();
+        BlobId blobIdCustom = testee().save(CUSTOM, SHORT_BYTEARRAY).block();
+
+        testee().delete(defaultBucketName, blobIdDefault).block();
+
+        assertThat(blobExistenceTester().exists(ObjectStorageBucketName.of(CUSTOM), blobIdCustom).block())
+            .isTrue();
+    }
+
+    @Test
+    default void deleteBucketShouldDeleteAllBlobIdsOfABucketInBlobExistenceTester() {
+        BucketName defaultBucketName = testee().getDefaultBucketName();
+
+        BlobId blobIdDefault = testee().save(defaultBucketName, SHORT_BYTEARRAY).block();
+        BlobId blobIdCustom = testee().save(defaultBucketName, OTHER_SHORT_BYTEARRAY).block();
+
+        testee().deleteBucket(defaultBucketName).block();
+
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(blobExistenceTester().exists(ObjectStorageBucketName.of(defaultBucketName), blobIdDefault).block())
+                .isFalse();
+            softly.assertThat(blobExistenceTester().exists(ObjectStorageBucketName.of(defaultBucketName), blobIdCustom).block())
+                .isFalse();
+        });
+    }
+
+    @Test
+    default void deleteBucketShouldDeleteAllBlobIdsWithDifferentBucketsInBlobExistenceTester() {
+        BucketName defaultBucketName = testee().getDefaultBucketName();
+
+        BlobId blobIdDefault = testee().save(defaultBucketName, SHORT_BYTEARRAY).block();
+        BlobId blobIdCustom = testee().save(CUSTOM, OTHER_SHORT_BYTEARRAY).block();
+
+        testee().deleteBucket(defaultBucketName).block();
+
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(blobExistenceTester().exists(ObjectStorageBucketName.of(defaultBucketName), blobIdDefault).block())
+                .isFalse();
+            softly.assertThat(blobExistenceTester().exists(ObjectStorageBucketName.of(CUSTOM), blobIdCustom).block())
+                .isFalse();
+        });
+    }
+
+    @Disabled("Failing around 0.3% of the time")
```

Comment:
Concurrent tests proves that mixed write+deletes can leave the application in an unstable state.